portfolio/src/main/webapp/script.js

```diff
@@ -27,24 +27,51 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-//deprecated
-function getQuoteOld() {
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('quote-container').innerText = quote;
-  });
-  console.log(response)
-  console.log(quote)
+
+function showComments() {
+    fetch('/data').then(response => response.json()).then((commentsJson) => {
+     
+        // unordered list
+        var ul = document.createElement('ul');
+
+        // where comments go
+        document.getElementById('comments-container').appendChild(ul);
+
+        // iterate through json object and make into html list
+        for (var key in commentsJson) {
+            let li = document.createElement('li');
+            li.innerHTML = commentsJson[key];
+            ul.appendChild(li);
+        }
+        // BUG: currently appends all comments each time "show comments" button is clicked
+    });
 }
 
-function getQuote() {
-    fetch('/data').then(response => response.json()).then((quotes) => {
-        // where the quote will go
-        quotesListElem = document.getElementById('quote-container');
+// redirect page to login upon "login/out" button click
+function loginButton() {
```

Comment:
Try to name functions as verbs, especially for callbacks. It's clearer to readers what onclick=login() or onclick=onLoginButtonClicked does rather than onclick=loginButton (login button what? hide it or log the user in or something else?)